Update `wagmi` to allow new chain support

dtbuchholz opened the pull request:
`wagmi` was outdated and unable to pull in Filecoin Calibration. Updating it required a few changes, including adding a `useSigner` hook, using a WalletConnect API key, and some type fixes due to `strict` compilation mode.

joewagner reviewed (dismissed):
@dtb This is a ton of great improvements!
I left an inline comment on a potential improvement to our use of `// @ts-ignore`, but it's not a reason to hold this up.

dtbuchholz commented on src/app/components/ContextProviders.tsx:
```diff
@@ -1,71 +1,53 @@
+import { Provider } from "react-redux";
+import { BrowserRouter } from "react-router-dom";
+import store from "../store/store";
+import GrandCentral from "./GrandCentral";
```
@joewagner nope, it wasn't set up, but i thought about adding them and should have! i'll open a separate PR and add those deps / clean up the things you noted below.

dtbuchholz commented:
hmm, i can mess with the git history and remove the unsigned commit / recreate it on my own, but is there a way around having to do that? the first commit isn't from me but an external collaborator i was working with, so i can't merge due to the fact it's unsigned.

dtbuchholz commented on src/app/store/tabsSlice.ts:
```diff
@@ -333,7 +331,6 @@ const tabsSlice = createSlice({
     builder.addCase(
       queryTableland.fulfilled,
-      // @ts-ignore
       (state, { payload: { columns, rows, query, tabId, error } }) => {
```
@joewagner i ended up adding linting....there was quite a lot to resolve! i've fixed everything but can't seem to figure out how to fix this last error. i'm guessing it's either the `undefined` part of the `type 'TablelandQueryResult | undefined'` that's expected, or it might be something with the fact `TablelandQueryResult` extends another type.

it complains with the following, in case you have any idea on how to resolve:
```md
ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts
./src/app/store/tabsSlice.ts 334:27-34
[tsl] ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts(334,28)
      TS2339: Property 'columns' does not exist on type 'TablelandQueryResult | undefined'.

ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts
./src/app/store/tabsSlice.ts 334:36-40
[tsl] ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts(334,37)
      TS2339: Property 'rows' does not exist on type 'TablelandQueryResult | undefined'.

ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts
./src/app/store/tabsSlice.ts 334:42-47
[tsl] ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts(334,43)
      TS2339: Property 'query' does not exist on type 'TablelandQueryResult | undefined'.

ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts
./src/app/store/tabsSlice.ts 334:49-54
[tsl] ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts(334,50)
      TS2339: Property 'tabId' does not exist on type 'TablelandQueryResult | undefined'.

ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts
./src/app/store/tabsSlice.ts 334:56-61
[tsl] ERROR in /Users/dtb/tbl/console/src/app/store/tabsSlice.ts(334,57)
      TS2339: Property 'error' does not exist on type 'TablelandQueryResult | undefined'.
```

dtbuchholz commented:
@joewagner i ended up added linting/prettier (there was a lot!) and fixed a bunch of other little things like the unsigned commit, removed React duplicate keys, etc. should be g2g.

joewagner reviewed (approved):
Nice work!  Lot's of improvements here
